src/java/fr/paris/lutece/plugins/forms/export/csv/FormResponseCsvExport.java

```diff
@@ -75,33 +59,43 @@ public FormResponseCsvExport( )
     {
         _csvSeparator = MultiviewConfig.getInstance( ).getCsvSeparator( );
     }
-
-    /**
-     * Build the CSV string for column line
+    /*
+     * Sort list of FormResponse by question export order
      */
+    public List<FormResponse> sortFormResponseListByQuestionOrder(List<FormResponse> formResponseList,  List<Question> listQuestions) {
+     List <FormResponse> sortedFormResponseList = new ArrayList<>();
+        HashMap<Integer, FormResponse> mapFormResponse = new HashMap<>();
+        for (int i = 0; i < listQuestions.size(); i++) {
+            Question question = listQuestions.get(i);
+          //  if (question.isResponseExportable()) {
+                for (FormResponse formResponse : formResponseList) {
+                    mapFormResponse.put(question.getExportDisplayOrder(), formResponse);
+            }
+         // }
+        }
+        for (Integer key : mapFormResponse.keySet()) {
+            sortedFormResponseList.add(mapFormResponse.get(key));
+        }
+        return sortedFormResponseList;
+    }
+        /**
+         * Build the CSV string for column line
+         */
     public String buildCsvColumnToExport( List<FormResponseItem> formResponseItems )
     {
     	List<FormResponse> formResponseList = getFormResponseFromItemList(formResponseItems);
     	int idForm = formResponseList.get(0).getFormId();
-        List<Step> listSteps = StepHome.getStepsListByForm( idForm );
-        List<Transition> listTransitions = TransitionHome.getTransitionsListFromForm( idForm );
-
-        List<Step> orderedStepList = StepService.sortStepsWithTransitions( listSteps, listTransitions );
+        List<Question> listQuestions = QuestionHome.getListQuestionByIdForm(idForm);
+        formResponseList = sortFormResponseListByQuestionOrder(formResponseList, listQuestions);
```

Comment:
FormResponses are sorted "by question order" , that means by qustion id, but just after this, the question list is sort by export order ?

Reply:
Yes, it was unclear. I've corrected the name of the function from "sortFormResponseListByQuestionOrder" to "sortFormResponseListByQuestionExportDisplayOrder". Users can set an export order for csv and pdf here /lutece/jsp/admin/plugins/forms/ManageFormExport.jsp?view=manageExport&id_form=1.  This is the order used in this function to sort the formResponse.